pkg/apis/core/types.go

```diff
@@ -3316,6 +3316,14 @@ type ServiceSpec struct {
 	// of peer discovery.
 	// +optional
 	PublishNotReadyAddresses bool
+
+	// topologyKeys is a preference-order list of topology keys which clients should respect when accessing this Service.
```

Comment:
nit: can you line-wrap comments at 80 columns?

Reply:
"which clients should respect" makes it sound like end-user clients are responsible for handling this themselves.
Also, nothing here defines what makes something a valid "topology key".